runtime/src/accounts.rs

```diff
@@ -872,7 +853,10 @@ impl Accounts {
                     last_blockhash_with_fee_calculator,
                     fix_recent_blockhashes_sysvar_delay,
                 );
-                let is_fee_payer = i == 0;
+                if fee_payer_index.is_none() {
+                    fee_payer_index = Some(i);
+                }
+                let is_fee_payer = Some(i) == fee_payer_index;
```

Comment:
:100:

Reply:
hmm, come to think of it, I think this logic change could also be abused to cause different behavior. Combined with this https://github.com/solana-labs/solana/pull/13868/files#r533407046, gating these doesn't make sense anymore?

Reply:
I believe getting an `i != 0` there would require someone to sneak around https://github.com/solana-labs/solana/blob/aebc3a17ce6f05d579ab3abb52eb813c56194e9e/sdk/program/src/message.rs#L203-L206 so they're probably already up to no good :slightly_smiling_face:

Reply:
yeah, it seems so. :)